snuba/views.py

```diff
@@ -349,7 +349,14 @@ def parse_and_run_query(dataset, request: Request, timer) -> QueryResult:
     with sentry_sdk.start_span(description=query.format_sql(), op="db") as span:
         span.set_tag("dataset", type(dataset).__name__)
         span.set_tag("table", source)
-        return raw_query(request, query, clickhouse_ro, timer, stats)
+        result = raw_query(request, query, clickhouse_ro, timer, stats)
+
+    with sentry_sdk.configure_scope() as scope:
+        if scope.span:
+            if "max_threads" in stats:
+                scope.span.set_tag("max_threads", stats["max_threads"])
```

Comment:
Why update this based on the value in `stats` rather than where it is set in `raw_query`?

Reply:
Because that would not be in the same span but in a child while I need this to be in the parent span (the transaction) otherwise it won't be indexed.
I could navigate back the tree from there but I suspect that's not the way tracing should work: you should be editing only your current span (which in general would be indexed).

Reply:
Makes sense, thanks for clarifying. 👍